examples/jsm/lights/RectAreaLightUniformsLib.js

```diff
@@ -37,8 +90,18 @@ var RectAreaLightUniformsLib = {
 
 		// data textures
 
-		var ltc_1 = new DataTexture( new Float32Array( LTC_MAT_1 ), 64, 64, RGBAFormat, FloatType, UVMapping, ClampToEdgeWrapping, ClampToEdgeWrapping, LinearFilter, NearestFilter, 1 );
-		var ltc_2 = new DataTexture( new Float32Array( LTC_MAT_2 ), 64, 64, RGBAFormat, FloatType, UVMapping, ClampToEdgeWrapping, ClampToEdgeWrapping, LinearFilter, NearestFilter, 1 );
+		const LTC_MAT_HALF_1 = new Uint16Array(LTC_MAT_1.length);
```

Comment:
I think it's better to only use half float when necessary. So as long as the platform supports `OES_texture_float_linear`, the data textures should use single precision floating point.

Reply:
OK I will make float32 the default, with float16 as a fallback option. 

This does mean that `RectAreaLightUniformsLib.init()` will need access to the renderer to test whether the `OES_texture_float_linear` extension is available. I'll start by simply adding `renderer` as an argument to `init()`, but this does change the API.

Reply:
Another option would be to generate both type of data textures in `RectAreaLightUniformsLib` but decide inside `WebGLRenderer` what type of texture to pick. In this way, the change would not break existing code. The check could be implemented here:

https://github.com/mrdoob/three.js/blob/1d5897874832f8f0fa5a2af3700597442fc28c31/src/renderers/webgl/WebGLLights.js#L408-L413

Reply:
Thanks this is a great suggestion. Just to clarify, for this approach we would need to bring the renderer into `WebGLLights`, e.g. via the constructors or setup functions for `WebGLRenderStates` and `WebGLLights`. Although this is changing function signatures, I believe this would only be changing the internal API.

Another option would be to add an extension check in `WebGLRenderer` itself:
https://github.com/mrdoob/three.js/blob/28364cd67ed79817e07eec3c8a39dd7a8b037410/src/renderers/WebGLRenderer.js#L1699-L1700

Reply:
It's not necessary to pass the entire renderer to `WebGLLights`. Just the `extensions` and `capabilities` object.

Reply:
I've implemented this approach in https://github.com/mrdoob/three.js/pull/20477/commits/ccfd940ede6dca8a36e605ada4b27994155d939d.